Does this build for you? I run into the following linker error for root:

```
     11843    [ 65%] Linking CXX shared library ../../lib/libRHTTP.so
     11844    cd /home/madlener/work/.spack/spack-stage/spack-stage-root-6.38.00-l4hkmvmotnbpzaczxjj6okj2bflsohcq/spack-build-l4hkmvm/net/http && /home/madlener/work/.spack/spackages/skylak
              e-ubuntu24.04-gcc13.3.0/cmake/3.31.9-6lz7nc/bin/cmake -E cmake_link_script CMakeFiles/RHTTP.dir/link.txt --verbose=1
     11845    /usr/bin/ld: /home/madlener/work/.spack/spackages/skylake-ubuntu24.04-gcc13.3.0/civetweb/1.16-6yp2po/lib/libcivetweb.a(civetweb.c.o): in function `send_compressed_data':
  >> 11846    /home/madlener/work/.spack/spack-stage/spack-stage-civetweb-1.16-6yp2poorcbbv5hybsguxkggotdss6rnw/spack-src/src/mod_zlib.inl:50:(.text+0xbeb4): undefined reference to `deflate
              Init2_'
  >> 11847    /usr/bin/ld: /home/madlener/work/.spack/spack-stage/spack-stage-civetweb-1.16-6yp2poorcbbv5hybsguxkggotdss6rnw/spack-src/src/mod_zlib.inl:83:(.text+0xbf78): undefined referenc
              e to `deflate'
  >> 11848    /usr/bin/ld: /home/madlener/work/.spack/spack-stage/spack-stage-civetweb-1.16-6yp2poorcbbv5hybsguxkggotdss6rnw/spack-src/src/mod_zlib.inl:62:(.text+0xc385): undefined referenc
              e to `deflateEnd'
  >> 11849    /usr/bin/ld: /home/madlener/work/.spack/spack-stage/spack-stage-civetweb-1.16-6yp2poorcbbv5hybsguxkggotdss6rnw/spack-src/src/mod_zlib.inl:123:(.text+0xc480): undefined referen
              ce to `deflateEnd'
  >> 11850    /usr/bin/ld: /home/madlener/work/.spack/spack-stage/spack-stage-civetweb-1.16-6yp2poorcbbv5hybsguxkggotdss6rnw/spack-src/src/mod_zlib.inl:71:(.text+0xc52a): undefined referenc
              e to `deflateEnd'
  >> 11851    collect2: error: ld returned 1 exit status
     11852    /home/madlener/work/.spack/spackages/skylake-ubuntu24.04-nonenone/compiler-wrapper/1.0-yhvkgq/libexec/spack/gcc/g++ -fPIC  -Wno-implicit-fallthrough -Wno-noexcept-type -pipe  
              -Wshadow -Wall -W -Woverloaded-virtual -fsigned-char -fsized-deallocation -pthread -O2 -g -DNDEBUG -Wl,--dependency-file=CMakeFiles/RHTTP.dir/link.d  -Wl,--no-undefined -Wl,--
              hash-style="both" -shared -Wl,-soname,libRHTTP.so.6.38 -o ../../lib/libRHTTP.so.6.38.00 CMakeFiles/RHTTP.dir/src/TCivetweb.cxx.o CMakeFiles/RHTTP.dir/src/TFastCgi.cxx.o CMakeF
              iles/RHTTP.dir/src/THttpCallArg.cxx.o CMakeFiles/RHTTP.dir/src/THttpEngine.cxx.o CMakeFiles/RHTTP.dir/src/THttpLongPollEngine.cxx.o CMakeFiles/RHTTP.dir/src/THttpServer.cxx.o 
              CMakeFiles/RHTTP.dir/src/THttpWSEngine.cxx.o CMakeFiles/RHTTP.dir/src/THttpWSHandler.cxx.o CMakeFiles/RHTTP.dir/src/TRootSniffer.cxx.o CMakeFiles/RHTTP.dir/src/TRootSnifferSto
              re.cxx.o CMakeFiles/G__RHTTP.dir/G__RHTTP.cxx.o  -Wl,-rpath,/home/madlener/work/.spack/spack-stage/spack-stage-root-6.38.00-l4hkmvmotnbpzaczxjj6okj2bflsohcq/spack-build-l4hkmv
              m/lib:/home/madlener/work/.spack/spackages/skylake-ubuntu24.04-gcc13.3.0/zlib-ng/2.2.4-edfclw/lib: ../../lib/libRIO.so.6.38.00 ../../lib/libThread.so.6.38.00 /home/madlener/wo
              rk/.spack/spackages/skylake-ubuntu24.04-gcc13.3.0/zlib-ng/2.2.4-edfclw/lib/libz.so -ldl -lcivetweb ../../lib/libCore.so.6.38.00
  >> 11853    make[2]: *** [net/http/CMakeFiles/RHTTP.dir/build.make:254: lib/libRHTTP.so.6.38.00] Error 1
     11854    make[2]: Leaving directory '/home/madlener/work/.spack/spack-stage/spack-stage-root-6.38.00-l4hkmvmotnbpzaczxjj6okj2bflsohcq/spack-build-l4hkmvm'
  >> 11855    make[1]: *** [CMakeFiles/Makefile2:56670: net/http/CMakeFiles/RHTTP.dir/all] Error 2
     11856    make[1]: *** Waiting for unfinished jobs....
```

[spack-build-out.txt](https://github.com/user-attachments/files/23832355/spack-build-out.txt)


I am wondering whether we run into any problems because ROOT actually vendors a newer version: https://github.com/root-project/root/pull/19136

--- a/repos/spack_repo/builtin/packages/civetweb/package.py
+++ b/repos/spack_repo/builtin/packages/civetweb/package.py
@@ -21,7 +21,7 @@ class Civetweb(CMakePackage):
     version("1.16", sha256="f0e471c1bf4e7804a6cfb41ea9d13e7d623b2bcc7bc1e2a4dd54951a24d60285")
 
     variant("shared", default=False, description="Build shared libraries instead of static ones")
-    
+
     depends_on("c", type="build")
     depends_on("cxx", type="build")
 

--- a/repos/spack_repo/builtin/packages/civetweb/package.py
+++ b/repos/spack_repo/builtin/packages/civetweb/package.py
@@ -13,7 +13,7 @@ class Civetweb(CMakePackage):
 
     homepage = "https://github.com/civetweb/civetweb"
     url = "https://github.com/civetweb/civetweb/archive/refs/tags/v1.16.tar.gz"
-
+    git = "https://github.com/civetweb/civetweb.git"
     maintainers("wdconinc")
 
     license("MIT", checked_by="wdconinc")

--- a/repos/spack_repo/builtin/packages/civetweb/package.py
+++ b/repos/spack_repo/builtin/packages/civetweb/package.py
@@ -20,6 +20,8 @@ class Civetweb(CMakePackage):
 
     version("1.16", sha256="f0e471c1bf4e7804a6cfb41ea9d13e7d623b2bcc7bc1e2a4dd54951a24d60285")
 
+    variant("shared", default=False, description="Build shared libraries instead of static ones")
+    
     depends_on("c", type="build")
     depends_on("cxx", type="build")
 
@@ -39,6 +41,7 @@ class Civetweb(CMakePackage):
             self.define("CIVETWEB_ENABLE_LUA", False),
             self.define("CIVETWEB_ENABLE_ZLIB", True),
             self.define("CIVETWEB_ENABLE_SSL", True),
+            self.define_from_variant("BUILD_SHARED_LIBS", "shared"),
         ]
         args.append(
             self.define("CIVETWEB_SSL_OPENSSL_API_1_0", self.spec.satisfies("^openssl@1.0"))

